### PR TITLE
New feature: Original system

### LINKED ISSUE #641: Keep objects' original system
Hello!

I have 2 systems: A and B. 
Whatever is developed in A is transported to B.
But whatever is developed in B is not transported anywhere.

Currently, when I use abapGit to send things from A to B... all the objects are created as if they were originally developed in B.

This is dangerous because abapGit is not the norm to keep the systems in sync and so it is possible (I'd say probable) that someone will make changes in system B to objects originally transported from system A. These changes will never be sent to system A.

This is dangerous.

It would be great if abapGit could read the original system of an object and then force into the objects created in other systems.

This could be an option at repository level.

Thanks,
Nuno

### PR DESCRIPTION
**Pull request by mbtools**
It's the longest outstanding issue from about 7 years ago and here's the solution to it.  Closes #641

You can now maintain the original system in your repo settings. This setting is then included in `.abapgit.xml`. When deploying such repository into a downstream system like Test or QA, abapGit with perform two addition steps:

1. Check the original system of existing objects

If an object has a different original system than the one in the repository, abapGit will not allow overwritting the object.

2. Update the original system after deserializing objects

Objects are set to the original system of the source. This will prevent changes to these objects.

Happy testing and Merry Christmas!

Example:

Settings:
image

After pull in system A4H:
image

Error in case system is different:
image

### CODE REVIEW

**Comment by larshp** [comment]
suggest using `TYPE tadir-srcsystem` instead of `TYPE srcsystem`, tadir is used in many places

then `SRCSYSTEM` can be added to [forbidden_void_types](https://rules.abaplint.org/forbidden_void_type/)

File: src/objects/zcl_abapgit_objects.clas.abap [hunk 1]
```diff
@@ -340,6 +354,35 @@ CLASS zcl_abapgit_objects IMPLEMENTATION.
...
+  METHOD check_original_system.
+
+    DATA:
+      lv_srcsystem TYPE srcsystem,
+      lv_error     TYPE abap_bool.
+
+    FIELD-SYMBOLS <ls_item> LIKE LINE OF it_items.
+
+    lv_srcsystem = io_dot->get_original_system( ).
+    IF lv_srcsystem IS NOT INITIAL.
```
Comment (fabianlupa) [inline src/objects/zcl_abapgit_objects.clas.abap:366]: Shouldn't it be `IS INITIAL`?

**Review by larshp — APPROVED** [review]
lgtm

**Comment by einareiriksson** [comment]
@mbtools been waiting for this feature for along time so I started testing. Works great with initial pull into a new system but releasing the transport request/task after gives an error becuase it's not of the correct type since now the objects should be in a repair transport becuase the original system has been changed after the objects were created. 

image

image

**Pull request merged**